src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Carnivore || herbivore.hidden) {
+      return;
+    } else {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health <= 0) {
+      const indexHerbivore = Animal.alive.findIndex(
+        (item) => item === herbivore,
+      );
+
+      Animal.alive.splice(indexHerbivore, 1);
+    }
+
+    return Animal.alive;
+  }
```

Comment:
The bite method should not return the Animal.alive array. This method should only be responsible for the biting action, not for returning the current state of all animals.